client/post-editor/editor-term-selector/index.jsx

```diff
@@ -26,21 +30,23 @@ class EditorTermSelector extends Component {
 	}
 
 	onTermsChange( selectedTerm ) {
-		const { postTerms, taxonomyName } = this.props;
+		const { postTerms, taxonomyName, siteId, postId } = this.props;
 		const terms = cloneDeep( postTerms ) || {};
-		const taxonomyTerms = terms[ taxonomyName ] || {};
 
-		if ( taxonomyTerms[ selectedTerm.name ] ) {
-			delete taxonomyTerms[ selectedTerm.name ];
+		// map call transforms object returned by API into an array
+		let taxonomyTerms = map( terms[ taxonomyName ] ) || [];
+
+		if ( find( taxonomyTerms, { ID: selectedTerm.ID } ) ) {
```

Comment:
To save a few cycles, maybe we should use a combination of `findIndex` and `Array#splice` instead of `find` and `reject`.